### DISCUSSION TITLE
How to read TimeValues from BaseIterativeData_t

### DISCUSSION THREAD

**Discussion by CedMoo** [CGNS/CGNS#710] [CGNStalk]
Hello,

We currently have some code reading CGNS files based on the CGIO API. There are however some performances issues when reading large files (or a large number of files) because the code is doing a lot of searching through the nodes of the file to find the appropriate node(s) to read.

I would like to make some experiments with the mid-level library to see if this would solve our performance issue and for this I'm currently investigating what I would need to implement in order to switch to the mid-level API. At some point, we need to be able to read the TimeValues from the BaseIterativeData_t node (see https://cgns.github.io/CGNS_docs_current/sids/timedep.html#BaseIterativeData). So far, I didn't find a way to do that in a straightforward way through the mid-level library. The only functions I found related to BaseIterativeData_t are the functions to read/write the number of steps (https://cgns.github.io/CGNS_docs_current/midlevel/timedep.html#biter). Did I miss something here?

Thanks.

**Comment by mennodeij** [comment]
You can find an example here, briefly it is reading arrays from `BaseIterativeData_t` node after a `cg_goto` call.

https://github.com/CGNS/CGNS/blob/a9aa8e632a24744c91370aa63967b04091854955/src/Test_UserGuideCode/C_code/read_timevert_str.c#L81

**Reply by CedMoo** [reply]
Thanks for your reply, this indeed works.
